scribus/scpaths.cpp

```diff
@@ -66,6 +67,35 @@ ScPaths::ScPaths() :
 	m_shareDir(SHAREDIR),
 	m_templateDir(TEMPLATEDIR)
 {
+
+// Build an appimage and redirect the paths to
+// ones inside the bundle, relative to the executable.
+#ifdef Q_OS_LINUX
+	// Set the application name expliticly.
+	QCoreApplication::setApplicationName("scribus");
+	QString pathPtr = QCoreApplication::applicationDirPath() + QString("/..");
+	m_shareDir = QString("%1/share/scribus/").arg(pathPtr);
+	m_docDir = QString("%1/share/doc/scribus/").arg(pathPtr);
+	//m_fontDir = QString("%1/share/scribus/fonts/").arg(pathPtr);
+	m_iconDir = QString("%1/share/scribus/icons/").arg(pathPtr);
+	m_sampleScriptDir = QString("%1/share/scribus/samples/").arg(pathPtr);
+	m_scriptDir = QString("%1/share/scribus/scripts/").arg(pathPtr);
+	m_templateDir = QString("%1/share/scribus/templates/").arg(pathPtr);
+	m_libDir = QString("%1/lib/scribus/").arg(pathPtr);
+	m_pluginDir = QString("%1/lib/scribus/plugins/").arg(pathPtr);
+	m_qmlDir = QString("%1/share/scribus/qml/").arg(pathPtr);
```

Comment:
All the `scribus` in the hardcoded paths there should use whatever C++ const is defined by the cmake variable `TAG_VERSION`.

Reply:
why should we add a version number in the folders?
would that not just add a new folder on every update?

Reply:
What is `TAG_VERSION` defined as.  By default it is an empty string, but whoever builds it can use it to be able to build different co-installable builds of scribus.

For example, for the Ubuntu PPA I'm providing 3 builds, 'scribus', 'scribus-ng' and 'scribus-trunk', all of them coinstallable.  For example, in the latest `TAG_VERSION` is `-trunk`.

Reply:
I don't know why *you* would be interested in adding a "number", but I am. :)

Reply:
Ok, I thought it was a value that was not empty by default...
how would I add the cmake value in the cpp code?